src/Tasks/CultureInfoCache.cs

```diff
@@ -35,6 +35,20 @@ static CultureInfoCache()
             {
                 ValidCultureNames.Add(cultureName.Name);
             }
+
+            // https://docs.microsoft.com/en-gb/windows/desktop/Intl/using-pseudo-locales-for-localization-testing
+            // These pseudo-locales are available in versions of Windows from Vista and later.
+            // However, from Windows 10, version 1803, they are not returned when enumerating the
+            // installed cultures, even if the registry keys are set. Therefore, add them to the list manually.
+            var pseudoLocales = new[] { "qps-ploc", "qps-ploca", "qps-plocm", "qps-Latn-x-sh" };
+
+            foreach (string pseudoLocale in pseudoLocales)
+            {
+                if (!ValidCultureNames.Contains(pseudoLocale))
+                {
+                    ValidCultureNames.Add(pseudoLocale);
```

Comment:
Since it's a `HashSet`, could you just do this unconditionally?

Actually, you could probably just do `ValidCultureNames.UnionWith(pseudoLocales)`. Though there's a `foreach` above so this being implemented the same way is nice.

Reply:
I was just being conservative in case a different (older) version of Windows to the one I'm running did return the pseudo locales and then threw an exception if it was already in there when it was added. Happy to amend if adding when already present would be safe.

Do you want me to retain the `foreach` or use the `UnionWith()`?

Reply:
`HashSet<T>.Add` can be called for the same item many times. Its [return value](https://docs.microsoft.com/en-us/dotnet/api/system.collections.generic.hashset-1.add?view=netframework-4.6) indicates whether it was actually added or already present, but we don't care in this case, so it's fine to ignore it. I prefer this because it's easier to read and reason about, and reduces indenting.

Let's just keep the `foreach`. It's consistent and evidently that's all `UnionWith` does anyway: https://github.com/dotnet/corefx/blob/58a310782b805df19a3939b12544c39296db5ed3/src/System.Collections/src/System/Collections/Generic/HashSet.cs#L507-L518

Reply:
Sorry, not quite sure from the comment. So do you just want me to drop the `Contains()`?

Reply:
Correct.

Reply:
Fixed in ce49c38a1caf238fda6a3e7ea14b112b8cf4836f.